crates/swc_ecma_transforms_module/src/path.rs

```diff
@@ -112,51 +112,13 @@ where
     R: Resolve,
 {
     fn resolve_import(&self, base: &FileName, module_specifier: &str) -> Result<JsWord, Error> {
-        fn to_specifier(
-            target_path: &str,
-            is_file: Option<bool>,
-            orig_ext: Option<&str>,
-        ) -> JsWord {
-            let mut p = PathBuf::from(target_path);
+        fn to_specifier(target_path: &str, orig_ext: Option<&str>) -> JsWord {
+            let p = PathBuf::from(target_path);
 
             if cfg!(debug_assertions) {
                 trace!("to_specifier: orig_ext={:?}", orig_ext);
             }
 
-            let dot_count = p
-                .file_name()
-                .map(|s| s.to_string_lossy())
-                .map(|v| v.as_bytes().iter().filter(|&&c| c == b'.').count())
-                .unwrap_or(0);
-
-            match orig_ext {
-                Some(orig_ext) => {
-                    if is_file.unwrap_or_else(|| p.is_file()) {
-                        if let Some(..) = p.extension() {
-                            if orig_ext == "ts"
-                                || orig_ext == "tsx"
-                                || orig_ext == "js"
-                                || orig_ext == "jsx"
-                                || dot_count == 1
-                            {
-                                p.set_extension(orig_ext);
-                            } else {
-                                p.set_extension("");
-                            }
-                        }
-                    }
-                }
-                _ => {
-                    if is_file.unwrap_or_else(|| p.is_file()) {
-                        if let Some(v) = p.extension() {
-                            if v == "ts" || v == "tsx" || v == "js" || v == "jsx" {
-                                p.set_extension("");
-                            }
-                        }
-                    }
-                }
-            }
```

Comment:
I think it was workaround for ts, i.e.
```
import foo from "./test.ts"
```

to

```
import foo from "./test"
```

So `test.js` can be resolved, not sure why it was implemented in this way :confused: 

babel keeps exetensions too

Reply:
🤣 I made lots of mistakes